File: binding/src/libqsequentialiterable.cpp
#include <QMetaSequence>
#include <QMetaType>
#include <QVariant>
#include <qsequentialiterable.h>
#include "libqsequentialiterable.h"
#include "libqsequentialiterable.hxx"

QVariantRef<QSequentialIterator> QSequentialIterator_OperatorMultiply(const QSequentialIterator* self) {
	return self->operator*();
}

QVariantPointer<QSequentialIterator> QSequentialIterator_OperatorMinusGreater(const QSequentialIterator* self) {
	return self->operator->();
}

void QSequentialIterator_OperatorAssign(QSequentialIterator* self, const QSequentialIterator* param1) {
	self->operator=(*param1);
}

void QSequentialIterator_Delete(QSequentialIterator* self) {
    delete self;
}

QVariant* QSequentialConstIterator_OperatorMultiply(const QSequentialConstIterator* self) {
	return new QVariant(self->operator*());
}

QVariantConstPointer* QSequentialConstIterator_OperatorMinusGreater(const QSequentialConstIterator* self) {
	return new QVariantConstPointer(self->operator->());
}

void QSequentialConstIterator_OperatorAssign(QSequentialConstIterator* self, const QSequentialConstIterator* param1) {
	self->operator=(*param1);
}

void QSequentialConstIterator_Delete(QSequentialConstIterator* self) {
    delete self;
}

QSequentialIterable* QSequentialIterable_new(const QSequentialIterable* other) {
	 return new QSequentialIterable(*other);
}

QSequentialIterable* QSequentialIterable_new2(QSequentialIterable* other) {
	 return new QSequentialIterable(*other);
}

QSequentialIterable* QSequentialIterable_new3() {
	 return new QSequentialIterable();
}

QSequentialIterable* QSequentialIterable_new4(const QMetaSequence* metaSequence, const QMetaType* metaType, void* iterable) {
	 return new QSequentialIterable(*metaSequence, *metaType, iterable);
}

QSequentialIterable* QSequentialIterable_new5(const QMetaSequence* metaSequence, const QMetaType* metaType, const void* iterable) {
	 return new QSequentialIterable(*metaSequence, *metaType, iterable);
}

QSequentialIterable* QSequentialIterable_new6(const QSequentialIterable* param1) {
	 return new QSequentialIterable(*param1);
}

void QSequentialIterable_CopyAssign(QSequentialIterable* self, QSequentialIterable* other) {
    *self = *other;
}

void QSequentialIterable_MoveAssign(QSequentialIterable* self, QSequentialIterable* other) {
    *self = std::move(*other);
}

QTaggedIterator<QSequentialConstIterator, void> QSequentialIterable_Begin(const QSequentialIterable* self) {
	return self->begin();
}

QTaggedIterator<QSequentialConstIterator, void> QSequentialIterable_End(const QSequentialIterable* self) {
	return self->end();
}

QTaggedIterator<QSequentialConstIterator, void> QSequentialIterable_ConstBegin(const QSequentialIterable* self) {
	return self->constBegin();
}

QTaggedIterator<QSequentialConstIterator, void> QSequentialIterable_ConstEnd(const QSequentialIterable* self) {
	return self->constEnd();
}

QTaggedIterator<QSequentialIterator, void> QSequentialIterable_MutableBegin(QSequentialIterable* self) {
	return self->mutableBegin();
}

QTaggedIterator<QSequentialIterator, void> QSequentialIterable_MutableEnd(QSequentialIterable* self) {
	return self->mutableEnd();
}

QVariant* QSequentialIterable_At(const QSequentialIterable* self, ptrdiff_t idx) {
	return new QVariant(self->at(idx));
}

void QSequentialIterable_Set(QSequentialIterable* self, ptrdiff_t idx, const QVariant* value) {
	self->set(idx, *value);
}

void QSequentialIterable_AddValue(QSequentialIterable* self, const QVariant* value) {
	self->addValue(*value);
}

void QSequentialIterable_RemoveValue(QSequentialIterable* self) {
	self->removeValue();
}

QMetaType* QSequentialIterable_ValueMetaType(const QSequentialIterable* self) {
	return new QMetaType(self->valueMetaType());
}

void QSequentialIterable_OperatorAssign(QSequentialIterable* self, const QSequentialIterable* param1) {
	self->operator=(*param1);
}

void QSequentialIterable_AddValue2(QSequentialIterable* self, const QVariant* value, int position) {
	self->addValue(*value, static_cast<QStyleOptionViewItem::Position>(position));
}

void QSequentialIterable_RemoveValue1(QSequentialIterable* self, int position) {
	self->removeValue(static_cast<QStyleOptionViewItem::Position>(position));
}

void QSequentialIterable_Delete(QSequentialIterable* self) {
    delete self;
}

File: binding/src/libqsequentialiterable.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qsequentialiterator.html
pub const qsequentialiterator = struct {

    pub fn OperatorMultiply(self: ?*anyopaque, ) QVariantRef<QSequentialIterator> {
        return qtc.QSequentialIterator_OperatorMultiply(@ptrCast(self));
    }

    pub fn OperatorMinusGreater(self: ?*anyopaque, ) QVariantPointer<QSequentialIterator> {
        return qtc.QSequentialIterator_OperatorMinusGreater(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSequentialIterator_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qsequentialconstiterator.html
pub const qsequentialconstiterator = struct {

    pub fn OperatorMultiply(self: ?*anyopaque, ) QtC.QVariant {
        return qtc.QSequentialConstIterator_OperatorMultiply(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSequentialConstIterator_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qsequentialiterable.html
pub const qsequentialiterable = struct {

    /// New constructs a new QSequentialIterable object.
    pub fn New() QtC.QSequentialIterable {
        return qtc.QSequentialIterable_new();
    }


    /// New2 constructs a new QSequentialIterable object.
    pub fn New2(metaSequence: ?*anyopaque, metaType: ?*anyopaque, iterable: ?*anyopaque) QtC.QSequentialIterable {
        return qtc.QSequentialIterable_new2(@ptrCast(metaSequence), @ptrCast(metaType), @ptrCast(iterable));
    }


    /// New3 constructs a new QSequentialIterable object.
    pub fn New3(metaSequence: ?*anyopaque, metaType: ?*anyopaque, iterable: ?*anyopaque) QtC.QSequentialIterable {
        return qtc.QSequentialIterable_new3(@ptrCast(metaSequence), @ptrCast(metaType), @ptrCast(iterable));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSequentialIterable_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSequentialIterable_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Begin(self: ?*anyopaque, ) QTaggedIterator<QSequentialConstIterator, void> {
        return qtc.QSequentialIterable_Begin(@ptrCast(self));
    }

    pub fn End(self: ?*anyopaque, ) QTaggedIterator<QSequentialConstIterator, void> {
        return qtc.QSequentialIterable_End(@ptrCast(self));
    }

    pub fn ConstBegin(self: ?*anyopaque, ) QTaggedIterator<QSequentialConstIterator, void> {
        return qtc.QSequentialIterable_ConstBegin(@ptrCast(self));
    }

    pub fn ConstEnd(self: ?*anyopaque, ) QTaggedIterator<QSequentialConstIterator, void> {
        return qtc.QSequentialIterable_ConstEnd(@ptrCast(self));
    }

    pub fn MutableBegin(self: ?*anyopaque, ) QTaggedIterator<QSequentialIterator, void> {
        return qtc.QSequentialIterable_MutableBegin(@ptrCast(self));
    }

    pub fn MutableEnd(self: ?*anyopaque, ) QTaggedIterator<QSequentialIterator, void> {
        return qtc.QSequentialIterable_MutableEnd(@ptrCast(self));
    }

    pub fn At(self: ?*anyopaque, idx: i64) QtC.QVariant {
        return qtc.QSequentialIterable_At(@ptrCast(self), idx);
    }

    pub fn Set(self: ?*anyopaque, idx: i64, value: ?*anyopaque) void {
        qtc.QSequentialIterable_Set(@ptrCast(self), idx, @ptrCast(value));
    }

    pub fn AddValue(self: ?*anyopaque, value: ?*anyopaque) void {
        qtc.QSequentialIterable_AddValue(@ptrCast(self), @ptrCast(value));
    }

    pub fn RemoveValue(self: ?*anyopaque, ) void {
        qtc.QSequentialIterable_RemoveValue(@ptrCast(self));
    }

    pub fn ValueMetaType(self: ?*anyopaque, ) QtC.QMetaType {
        return qtc.QSequentialIterable_ValueMetaType(@ptrCast(self));
    }

    pub fn AddValue2(self: ?*anyopaque, value: ?*anyopaque, position: i32) void {
        qtc.QSequentialIterable_AddValue2(@ptrCast(self), @ptrCast(value), @intCast(position));
    }

    pub fn RemoveValue1(self: ?*anyopaque, position: i32) void {
        qtc.QSequentialIterable_RemoveValue1(@ptrCast(self), @intCast(position));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSequentialIterable_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/sequentialiterable.html#types
pub const enums = struct {
    pub const Position = enum {
        pub const Unspecified: i32 = 0;
        pub const AtBegin: i32 = 1;
        pub const AtEnd: i32 = 2;
    };

};
